core/src/impl/Kokkos_HostThreadTeam.hpp

```diff
@@ -310,8 +308,9 @@ class HostThreadTeamData {
   //   thread_local_size = number bytes for thread local memory
   // Return:
   //   total number of bytes that must be allocated
-  static size_t scratch_size(int pool_reduce_size, int team_reduce_size,
-                             int team_shared_size, int thread_local_size) {
+  static size_t scratch_size(size_t pool_reduce_size, size_t team_reduce_size,
+                             size_t team_shared_size,
+                             size_t thread_local_size) {
```

Comment:
Similar question why changing the type of `thread_local_size`?

Reply:
Just for consistency. Especially on the host, I would expect `thread_local_size` possibly to be of similar magnitude as `team_shared_size`.